### DISCUSSION TITLE
How to create custom non-eloquent type that is linked to Eloquent type

### DISCUSSION THREAD

**Discussion by WordsofDefiance** [nuwave/lighthouse#2541] [Q&A]
I have a model, User. This user has many Policies, which my app can only access via an API.

It's easy enough to create a custom query that calls the API and returns the type:

```
type User {
   id 
   name
}

type Policy {
   id
   product_number
}
```

```php
<?php

declare(strict_types=1);

namespace App\GraphQL\Queries;

class Product
{
    private API $api;

    public function __construct(API $api)
    {
        $this->api = $api;
    }

    /**
     * @param  null  $_
     * @param  array<string, mixed>  $filters
     */
    public function __invoke($_, array $filters): array
    {
        $response = $this->api->request('/api/v1/products', 'GET', $filters);

        return $response;
    }
}
```

But how can I link this to the user model without having an explicitly defined `hasMany` relationship? By using a type resolver? I feel like I could build a model for a `Product`, but that feels janky since there's no actual table that the model would be linked to.

the query would look something like this:

```
{
  user {
    id
    name
    product {
      product_number
    }
  }
}
```

Thanks!

**Comment by spawnia** [comment]
You don't have to build a model. As long as the shape returned by the API matches your GraphQL types and is either an associative array or a plain object with public properties, you can return it as is. If not, you can wrap it in a value object.
